sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -628,7 +628,17 @@ public void init(CosmosClientMetadataCachesSnapshot metadataCachesSnapshot, Func
                     this.clientTelemetryConfig,
                     this,
                     this.connectionPolicy.getPreferredRegions());
-            clientTelemetry.init();
+            clientTelemetry.init().doOnTerminate(() -> {
```

Comment:
I am curious why are we using `doOnTerminate()` and not `doOnSubscribe()`? Since this is for side effect, they suggest to add monitoring to doOnSubscribe()

Reply:
doOnSubscribe would trigger before the step is completed - but  changed it to thenxxx as well.